test/e2e/network/loadbalancer.go

```diff
@@ -1236,6 +1244,134 @@ var _ = common.SIGDescribe("LoadBalancers", func() {
+		}
+		ds.Spec.Template.Labels = labels
+
+		nodeNames := e2edaemonset.SchedulableNodes(cs, ds)
+		e2eskipper.SkipUnlessAtLeast(len(nodeNames), 2, "load-balancer rolling update test requires at least 2 schedulable nodes for the DaemonSet")
+
+		ginkgo.By(fmt.Sprintf("Creating DaemonSet %q", name))
+		ds, err = cs.AppsV1().DaemonSets(ns).Create(context.TODO(), ds, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Checking that daemon pods launch on every schedulable node of the cluster")
+		err = wait.PollImmediate(dsRetryPeriod, dsRetryTimeout, e2edaemonset.CheckDaemonPodOnNodes(f, ds, nodeNames))
+		framework.ExpectNoError(err, "error waiting for daemon pods to start")
+		err = e2edaemonset.CheckDaemonStatus(f, name)
+		framework.ExpectNoError(err)
+
+		ginkgo.By(fmt.Sprintf("Creating a service %s with type=LoadBalancer in namespace %s", name, ns))
+		jig := e2eservice.NewTestJig(cs, ns, name)
+		jig.Labels = labels
+		service, err := jig.CreateLoadBalancerService(e2eservice.GetServiceLoadBalancerCreationTimeout(cs), func(svc *v1.Service) {})
+		framework.ExpectNoError(err)
+
+		lbNameOrAddress := e2eservice.GetIngressPoint(&service.Status.LoadBalancer.Ingress[0])
+		svcPort := int(service.Spec.Ports[0].Port)
+
+		ginkgo.By("Hitting the DaemonSet's pods through the service's load balancer")
+		timeout := e2eservice.LoadBalancerLagTimeoutDefault
+		if framework.ProviderIs("aws") {
+			timeout = e2eservice.LoadBalancerLagTimeoutAWS
+		}
+		e2eservice.TestReachableHTTP(lbNameOrAddress, svcPort, timeout)
+
+		ginkgo.By("Starting a goroutine to continuously hit the DaemonSet's pods through the service's load balancer")
+		var totalRequests uint64 = 0
+		var failedRequests uint64 = 0
+		done := make(chan struct{})
+		defer close(done)
+		go func() {
+			defer ginkgo.GinkgoRecover()
+
+			wait.Until(func() {
+				atomic.AddUint64(&totalRequests, 1)
+				client := &http.Client{
+					Transport: utilnet.SetTransportDefaults(&http.Transport{
+						DisableKeepAlives: true,
+					}),
+					Timeout: 5 * time.Second,
+				}
+				ipPort := net.JoinHostPort(lbNameOrAddress, strconv.Itoa(svcPort))
+				msg := "hello"
+				url := fmt.Sprintf("http://%s/echo?msg=%s", ipPort, msg)
+				resp, err := client.Get(url)
+				if err != nil {
+					framework.Logf("Got error testing for reachability of %s: %v", url, err)
+					atomic.AddUint64(&failedRequests, 1)
+					return
+				}
+				defer resp.Body.Close()
+				if resp.StatusCode != http.StatusOK {
+					framework.Logf("Got bad status code: %d", resp.StatusCode)
+					atomic.AddUint64(&failedRequests, 1)
+					return
+				}
+				body, err := io.ReadAll(resp.Body)
+				if err != nil {
+					framework.Logf("Got error reading HTTP body: %v", err)
+					atomic.AddUint64(&failedRequests, 1)
+					return
+				}
+				if string(body) != msg {
+					framework.Logf("The response body does not contain expected string %s", string(body))
+					atomic.AddUint64(&failedRequests, 1)
+					return
+				}
```

Comment:
should we differentiate between network errors and http errors?

if err != nil is a network error [per the golang std library docs](https://pkg.go.dev/net/http#Client.Do) and the rest request or http errors , something like that

Reply:
You are suggesting to only count network errors ? Or both ? Should we check success rate differently if we count both ?

All the errors encountered so far, in my testing, are network errors.

Reply:
count both for reporting, in case the test fail I'd like to check what amount are network errors and what are http errors, but we can keep using the aggragation for the threshold

Reply:
PR updated.